src/projection.jl

```diff
@@ -30,6 +30,17 @@ function projection(C::IsometricKroneckerCovariance{elType}) where {elType}
     return Proj
 end
 
+function projection(C::BlockDiagonalCovariance{elType}) where {elType}
+    Proj(deriv) = begin
+        e_i = zeros(elType, C.q + 1, 1)
+        if deriv <= C.q
+            e_i[deriv+1] = 1
+        end
+        return MFBD([e_i' for _ in 1:C.d])
```

Comment:
Should probably use a `copy` too, even though this is probably never mutated in the code here.